PURL-SPECIFICATION.rst

```diff
@@ -222,6 +222,8 @@ The rules for each component are:
     - MUST NOT contain a '/'
     - MUST NOT be any of '..' or '.'
     - MUST NOT be empty
+    - MAY contain any Unicode character unless this ``subpath`` definition or
```

Comment:
and these unicode characters are still to be url-encoded?

Reply:
We should probably define a **standard set of forbidden characters** for `subpath` fields across PURL types.

Characters like `<`, `>`, `:`, `"`, `/`, `\`, `|`, `?`, and `*` are invalid on NTFS and other Windows file systems, so allowing them would harm portability.

Currently, only `golang` and `cocoapods` use `subpath`:

* Go explicitly forbids these characters (see [golang/go#45549](https://github.com/golang/go/issues/45549)).
* CocoaPods subpaths must follow even stricter C99 identifier rules.

To ensure consistency and allow room for future extensions, we should **reserve these 8 characters** globally (beyond the existing `/` restriction)?

Reply:
Now, there are so many possible constraints from OS, filesystem and package type that we cannot specify sanely a set of characters or restrictions that would work everywhere. This is why the subpath is relative to the root of a package, and using whatever characters are used there. There will be a few rare corner cases that may feel weird and will be weird across some OSes.... but let these be weird. If this ever comes up as a serious problem, we can address it then.